com/metacube/DSA/Assignment3/VirtualCommandPrompt.java

```diff
@@ -0,0 +1,252 @@
+package GET2018.com.metacube.DSA.Assignment3;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+/**
+ * Copyright (c) 2018 Metacube.com. All rights reserved.
+ * This class lets user give commands to virtual command prompt.
+ * @author Chirag Jain
+ * 
+ */
+public class VirtualCommandPrompt {
+
+	private static List<Directory> listOfDirectory = new ArrayList<Directory>();
+	private static String path = ".\\";
+	
+	/**
+	 * This class takes input from user and give results according to command.
+	 */
+	public static void virtualCommandPrompt()
+	{
+		Directory root = new Directory("R:", new Date());
+		listOfDirectory.add(root);
+		@SuppressWarnings("resource")
+        Scanner scanner = new Scanner(System.in);
+		System.out.println("***Welcome to Virtual Command Prompt***");
+		String signature = "R:\\";
+		System.out.print(signature + ">");
+		Directory currentDirectory = root;
+		
+		while(true)
+		{
+			String commandLine = scanner.nextLine();
+			String[] commandWords = commandLine.split(" ");
+			
+			switch(commandWords[0])
+			{
+			
+			case "mkdir" : 			    
+			    try
+                {
+			        if(commandWords.length != 2) {
+			            throw new CustomException("Invalid use of command mkdir");
+			        }
+			         
+			        Directory newDirectory = new Directory(commandWords[1], new Date());
+			        listOfDirectory.add(newDirectory);
+                    currentDirectory.getListOfSubDirectory().add(newDirectory);                                            
+                } catch (CustomException e)
+                {
+                    System.out.println(e.getMessage());
+                }
+			    System.out.print(signature + ">");
+			  	break;
+						   
+			case "cd"    : 			    
+			    try
+                {
+                    if(commandWords.length != 2) {
+                        throw new CustomException("Invalid use of command cd");
+                    }
+                     
+                    if(search(currentDirectory, commandWords[1]))
+                    {
+                         for(Directory directory: listOfDirectory)
+                         {
+                             if(directory.getName().equals(commandWords[1]))
+                             {
+                                 currentDirectory = directory;
```

Comment:
Don't use this many for loops, they increase processing time and is not good practice. What you can do here is first check if the command[i] is blank or null. After that use list methods for processing: 

if(!currentDirectory.getSubDirectoryList().isEmpty() && currentDirectory.getSubDirectorylist.contains(commandWords[i]){
	//get the word
	int index = currentDirectory.getSubDirectoryList().lastIndexOf(commandWords[i]);
	string signature = currentDirectory.getSubDirectoryList().get(index);
 }
I hope this helps.